### PR TITLE
Incremental update

### PR DESCRIPTION
**Pull request by feiranwang**
1. Update incremental doc, fixing typos etc.
2. Update incremental spouse example, reflecting the latest 
3. Update scripts for running incremental spouse example, using deepdive commands

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -30,3 +30,4 @@
 .mindtagger.conf.json
 /out
 /log
+/lib
```
Comment (netj) [inline .gitignore]: Why is this added back?

File: examples/spouse_example/postgres/incremental/setup_database.sh [hunk 2]
```diff
@@ -12,9 +12,11 @@ fi
...
+# dropdb $DBNAME || true
+# createdb $DBNAME
 
-psql -d $DBNAME <./schema.sql
+# psql -d $DBNAME <./schema.sql
 bzcat ../../data/articles_dump.csv.bz2  | psql -d $DBNAME -c "COPY articles  FROM STDIN CSV"
-bzcat ../../data/sentences_dump.csv.bz2 | psql -d $DBNAME -c "COPY sentences FROM STDIN CSV"
+bzcat ../../data/sentences_dump.csv.bz2 |
+if [[ -z ${INCREMENTAL_SPOUSE_EXAMPLE_LIMIT_SENTENCES:-} ]]; then cat; else head -n ${INCREMENTAL_SPOUSE_EXAMPLE_LIMIT_SENTENCES}; fi |
+psql -d $DBNAME -c "COPY sentences FROM STDIN CSV"
```
Comment (netj) [inline examples/spouse_example/postgres/incremental/setup_database.sh]: Let's use `deepdive sql`.

File: examples/spouse_example/postgres/incremental/spouse_example.f1.active.rules [hunk 3]
```diff
@@ -1,3 +1,3 @@
 # List names of the inference rule to mark as active in materialization phase here.
 # It's in the form of the relation name of the head in the inference rule followed by an underscore and the zero-based index of the rule.
-has_spouse_0  # You can put comments after a hash letter.
+inference_rule_0  # You can put comments after a hash letter.
```
Comment (netj) [inline examples/spouse_example/postgres/incremental/spouse_example.f1.active.rules]: This is another place where the compiler naming convention needs to be fixed as pointed out in #411.

File: shell/deepdive-run [hunk 4]
```diff
@@ -73,6 +73,7 @@ else
     run_dir="$DEEPDIVE_APP/run/$run_id"
 fi
 mkdir -p "$run_dir"
+mkdir -p run
```
Comment (netj) [inline shell/deepdive-run]: Is this necessary?  If so, let's add a comment.

File: .gitignore (hunk 1 above)
Comment (feiranwang) [inline .gitignore]: I was thinking we don't want to track the library files. `library/` is also not included in the source tree, so we are currently not tracking it.

**Comment by netj** [comment]
Documentation update looks good.
A few high-level comments:
* Ideally, we should get rid of run.sh, and have `deepdive run` accept a phase (mode) option that gets passed down to `ddlog compile`.
* Most of setup_database.sh lines should be moved to input/init.sh, and `deepdive initdb` should be used instead of `createdb` in 0-setup.sh.

File: .gitignore (hunk 1 above)
Comment (netj) [inline .gitignore]: But do any of our source or build produce `lib/` or `library/`?  I don't think we need to over ignore.

Comment (feiranwang) [inline .gitignore]: I see. It's probably old files on my laptop. I'll remove this.

**Comment by feiranwang** [comment]
Updated and rebased so that the `.gitignore` change is not included.
